### PR TITLE
Add Benchmark guide to doc

### PR DESCRIPTION
**Pull request by duc-nx**
Summary:

Add Benchmark.md up to 13-th Fibonacci  number.

### CODE REVIEW

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (yoichi-nexus) [inline docs/pages/zkvm/sdk-benchmarking.mdx:43]: nit: this looks easier, but I haven't tried myself
```suggestion
    prover.prove(pp).expect("failed to prove program")
```

Comment (duc-nx) [reply docs/pages/zkvm/sdk-benchmarking.mdx:43]: Thanks for the review @yoichi-nexus , I intend to name the return variable as `proof` so people acknowledge the outcome of a prove API is a proof. Your suggestion definitely works. What do you think?

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (yoichi-nexus) [inline docs/pages/zkvm/sdk-benchmarking.mdx:14]: Vercel is rendering three numbered steps: "RISC-V Cycles count" "Execution time" "Graphs". Did you want these three steps?

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (yoichi-nexus) [inline docs/pages/zkvm/sdk-benchmarking.mdx:16]: I clicked "SDK Benchmarking" and came here. Now I'm confused because I didn't see anything about benchmarking in the first paragraph.

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (duc-nx) [reply docs/pages/zkvm/sdk-benchmarking.mdx:14]: Yup. I want to show the throughout benchmark has 3 parts, the storyline is:
1. At guest side: Code can be profiled base on the number of RISC-V cycles
2. At host side: Proving time can be profiled. 
3. The combine results of 1 and 2. 
What do you think?

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (duc-nx) [reply docs/pages/zkvm/sdk-benchmarking.mdx:16]: Oh yeah, I should update this line.

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (yoichi-nexus) [reply docs/pages/zkvm/sdk-benchmarking.mdx:43]: no strong opinons. It's like a soft-clippy comment.

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (yoichi-nexus) [reply docs/pages/zkvm/sdk-benchmarking.mdx:14]: Those three steps make sense, so there will be headlines called something like
* guest side benchmark
* host side benchmark
* combining guest & host benchmarks

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (yoichi-nexus) [inline docs/pages/zkvm/sdk-benchmarking.mdx:79]: An explanation about `#[nexus_rt::profile]` helps readers who are interested in benchmarking.

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (yoichi-nexus) [inline docs/pages/zkvm/sdk-benchmarking.mdx:18]: Are we following the file layout generated by SDK quick start?

Do we run this command from the SDK quick start?
```
cargo install --git https://github.com/nexus-xyz/nexus-zkvm cargo-nexus --tag 'v0.2.1'
```
Do we do something else?

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (yoichi-nexus) [inline docs/pages/zkvm/sdk-benchmarking.mdx:123]: Does this table come from `#[nexus_rt::profile]` in the guest program?

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (sjudson) [inline docs/pages/zkvm/sdk-benchmarking.mdx:16]: This bit should come before the steps start.

File: docs/pages/zkvm/_meta.json
Comment (sjudson) [inline docs/pages/zkvm/_meta.json:5]: I would remove "SDK" from the title. It can be inside the document, but I think it's unnecessary from a title perspective.

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (sjudson) [reply docs/pages/zkvm/sdk-benchmarking.mdx:16]: > The example uses the `guest` package as an example guest program.

This sentence comes across as confusing and tautological. Maybe rename the package from `guest` to `example` or something?

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (sjudson) [reply docs/pages/zkvm/sdk-benchmarking.mdx:18]: Maybe

> The `src/main.rs` at host side is as follows:

could be

> The host program `src/main.rs` is:

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (sjudson) [inline docs/pages/zkvm/sdk-benchmarking.mdx:41]: Why not just import `Proof`?

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (sjudson) [reply docs/pages/zkvm/sdk-benchmarking.mdx:123]: Yeah we should explain the provenance.

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (sjudson) [inline docs/pages/zkvm/sdk-benchmarking.mdx:111]: How long is the list of supported CPUs? Should we enumerate them?

**Review by sjudson — CHANGES_REQUESTED** [review]

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (yoichi-nexus) [inline docs/pages/zkvm/sdk-benchmarking.mdx:12]: The first heading in this page should match the title on line 2.

Comment (duc-nx) [reply docs/pages/zkvm/sdk-benchmarking.mdx]: Fixed in [36f79a4](https://github.com/nexus-xyz/nexus-zkvm/pull/271/commits/36f79a4ffc509fe44f4c288746c674652d698a4b)

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (duc-nx) [reply docs/pages/zkvm/sdk-benchmarking.mdx]: I updated in [36f79a4](https://github.com/nexus-xyz/nexus-zkvm/pull/271/commits/36f79a4ffc509fe44f4c288746c674652d698a4b), I switch `target-cpu` to `native` and provide instruction to set a specific CPU.

File: docs/pages/zkvm/_meta.json
Comment (duc-nx) [reply docs/pages/zkvm/_meta.json]: Updated in [36f79a4](https://github.com/nexus-xyz/nexus-zkvm/pull/271/commits/36f79a4ffc509fe44f4c288746c674652d698a4b)

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (duc-nx) [reply docs/pages/zkvm/sdk-benchmarking.mdx]: I replaced the entire paragraph with new one simpler: 

```
This benchmark demonstrates the performance of Nexus zkVM using a Fibonacci sequence calculation as an example. We'll analyze both the guest program's RISC-V cycle count and the host program's execution time for various Fibonacci sequence lengths.
```

Updated in [36f79a4](https://github.com/nexus-xyz/nexus-
zkvm/pull/271/commits/36f79a4ffc509fe44f4c288746c674652d698a4b)

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (duc-nx) [reply docs/pages/zkvm/sdk-benchmarking.mdx]: updated in [36f79a4](https://github.com/nexus-xyz/nexus-zkvm/pull/271/commits/36f79a4ffc509fe44f4c288746c674652d698a4b)

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (duc-nx) [reply docs/pages/zkvm/sdk-benchmarking.mdx]: updated in [36f79a4](https://github.com/nexus-xyz/nexus-zkvm/pull/271/commits/36f79a4ffc509fe44f4c288746c674652d698a4b)

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (duc-nx) [reply docs/pages/zkvm/sdk-benchmarking.mdx:124]: Explained in [36f79a4](https://github.com/nexus-xyz/nexus-zkvm/pull/271/commits/36f79a4ffc509fe44f4c288746c674652d698a4b)

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (duc-nx) [reply docs/pages/zkvm/sdk-benchmarking.mdx]: Ah, I did tried, the compiler need me to specify the return type of function, so I put the full path.

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (yoichi-nexus) [inline docs/pages/zkvm/sdk-benchmarking.mdx:11]: All the other pages start with
```
## Title of the page
```
So let's follow the pattern here too.

Comment (duc-nx) [reply docs/pages/zkvm/sdk-benchmarking.mdx:11]: updated in a61c6da

Comment (yoichi-nexus) [reply docs/pages/zkvm/sdk-benchmarking.mdx:11]: In all the other pages, the we have only one `##` heading. The rest is `###` and more #'s.

**Comment by yoichi-nexus** [comment]
I left comments in Vercel. Especially the layout looks odd.

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (duc-nx) [reply docs/pages/zkvm/sdk-benchmarking.mdx]: fixed in 88d1d92

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (duc-nx) [reply docs/pages/zkvm/sdk-benchmarking.mdx]: @yoichi-nexus  I think there is a page dedicated for that so I don't mention it here. 
@sjudson Updated in the latest commit.

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (sjudson) [inline docs/pages/zkvm/sdk-benchmarking.mdx:31]: If there's no reason for the extra memory over the 4mb base then I'd remove this and rerun the benchmarks (they're unnecessarily slow).

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (sjudson) [inline docs/pages/zkvm/sdk-benchmarking.mdx:94]: Unnecessary space.

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (sjudson) [inline docs/pages/zkvm/sdk-benchmarking.mdx:109]: grammar: "specific CPUs" -> "specific CPU"

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (sjudson) [inline docs/pages/zkvm/sdk-benchmarking.mdx:110]: grammar : "you can obtain the list of target CPUs can be obtained from this command" -> "the list of available target CPUs can be obtained with the command"

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (sjudson) [inline docs/pages/zkvm/sdk-benchmarking.mdx:121]: grammar: "are obtained" -> "were obtained"

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (sjudson) [inline docs/pages/zkvm/sdk-benchmarking.mdx:125]: grammar: "Run command at `src/guest/` directory:" -> "In the `src/guest/` directory, run the command:"

Comment (sjudson) [reply docs/pages/zkvm/sdk-benchmarking.mdx:125]: Also, shouldn't this just be `cargo nexus ...`, here and below?

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (sjudson) [inline docs/pages/zkvm/sdk-benchmarking.mdx:209]: I'd cut this sentence.

**Review by sjudson — CHANGES_REQUESTED** [review]
A few more comments, but it's getting there.

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (yoichi-nexus) [reply docs/pages/zkvm/sdk-benchmarking.mdx]: > I think there is a page dedicated for that

Please add a link to that page. That makes the benchmark more easily reproducible.

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (duc-nx) [reply docs/pages/zkvm/sdk-benchmarking.mdx]: updated in 915ef4c

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (duc-nx) [reply docs/pages/zkvm/sdk-benchmarking.mdx]: updated in 915ef4c

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (duc-nx) [reply docs/pages/zkvm/sdk-benchmarking.mdx]: updated in 915ef4c

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (duc-nx) [reply docs/pages/zkvm/sdk-benchmarking.mdx]: updated in 915ef4c

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (duc-nx) [reply docs/pages/zkvm/sdk-benchmarking.mdx]: updated in 915ef4c

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (duc-nx) [reply docs/pages/zkvm/sdk-benchmarking.mdx]: updated in 915ef4c

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (yoichi-nexus) [inline docs/pages/zkvm/sdk-benchmarking.mdx:184]: Do we miss a lot if we cut these lines out?

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (duc-nx) [reply docs/pages/zkvm/sdk-benchmarking.mdx]: grammar: updated in 915ef4c . 

The `cargo nexus run` is to get the trace cycle here.

For proving and verifying, I follow the SDK, at the `host`, a `cargo run -r` at root directory is enough. 
Otherwise it will be `cargo nexus prove` inside `/src/guest/` directory.  I prefer the SDK approach, cleaner.

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (duc-nx) [reply docs/pages/zkvm/sdk-benchmarking.mdx:184]: If no objection, I want to keep these lines.

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (sjudson) [reply docs/pages/zkvm/sdk-benchmarking.mdx]: The complaint is solely about using `cargo-nexus`, as opposed to `cargo nexus ...`, which is a cleaner UI for the same functionality.

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (yoichi-nexus) [reply docs/pages/zkvm/sdk-benchmarking.mdx:184]: Okay.

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (duc-nx) [reply docs/pages/zkvm/sdk-benchmarking.mdx]: I see. Updated in efdec84 . 
Also I removed the `#Build Flags` section, assuming user has followed the `SDK_quickstart.mdx`.

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (duc-nx) [reply docs/pages/zkvm/sdk-benchmarking.mdx]: updated in in c8ed3c3  at the first section

**Review by sjudson — APPROVED** [review]
Looks good to me, but I'd wait on @yoichi-nexus's approval as well before merging

File: docs/pages/zkvm/sdk-benchmarking.mdx
Comment (yoichi-nexus) [reply docs/pages/zkvm/sdk-benchmarking.mdx]: The instruction seems incomplete. I left a comment in Vercel. I'm closing this thread.

**Review by yoichi-nexus — CHANGES_REQUESTED** [review]
The diff is needed to run the benchmark.

**Comment by duc-nx** [comment]
The doc works after 0.2.3

**Comment by yoichi-nexus** [comment]
> The doc works after 0.2.3

No it doesn't. After following the doc, I didn't get any `Cargo.toml`.

**Comment by yoichi-nexus** [comment]
I'm following the doc again.

**Comment by duc-nx** [comment]
Okay, I updated the doc which contain the instruction to generate project and show the Cargo.toml in 26e913a

**Review by yoichi-nexus — CHANGES_REQUESTED** [review]
I left a comment in Vercel. `cargo run --release 2>/dev/null` doesn't print anything. I could not find performance numbers. Probably the document should explain how to see the performance numbers.

**Review by yoichi-nexus — CHANGES_REQUESTED** [review]
I'm seeing a compilation error
```
  --> src/main.rs:18:53
   |
18 | fn prove_execution(pp: &PP, prover: Nova<Local>) -> nexus_sdk::nova::seq::Proof {
   |    ---------------                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `Proof`, found `()`
   |    |
   |    implicitly returns `()` as its body has no tail or `return` expression
19 |     prover.prove(pp).expect("failed to prove program");
   |                                                       - help: remove this semicolon to return this value
```

**Comment by duc-nx** [comment]
It should work now. I left a typo after modifying the code according to the soft clippy suggestion. ac87b79

**Review by yoichi-nexus — APPROVED** [review]
Now it's working for me!

**Pull request merged**
